pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -607,19 +665,43 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 		return actionContinue{provResult.RequeueAfter}
 	}
 
+	pendingRebootSince := info.host.Status.Provisioning.PendingRebootSince
+	poweredOnAt := info.host.Status.Provisioning.PoweredOnAt
+
+	desiredPowerOnState := info.host.Spec.Online
+	isInRebootProcess := poweredOnAt.Before(pendingRebootSince) || (!pendingRebootSince.IsZero() && poweredOnAt.IsZero())
+
+	if isInRebootProcess {
+		if !info.host.Status.PoweredOn {
```

Comment:
It's not clear to me why this is nested inside `if isInRebootProcess`. If we have a suffixless annotation present and the power is off, we should always just clear it I think?

Reply:
If we check the provisioning state on line 654 then we will still need to move this outside that block.